### PR TITLE
[REVIEW] adding model saving, serving docs, and results analysis

### PR DESCRIPTION
**Pull request by miroenev**
- Added best model download from S3
- Added results analysis details and best-job query
- Added details about deploying/inference
- Added safeguard against https://github.com/rapidsai/cudf/issues/5579
- Removed dataset repartition/rebalance [ was causing RF failure issues ]
- Contributed feedback to https://github.com/dask/distributed/issues/3955 RE dask.nanny issues
- Removed /notebook directory and surfaced notebook to the base directory

### CODE REVIEW

**Comment by miroenev** [comment]
I renamed the notebook to rapids_sagemaker_hpo.ipynb instead of sagemaker_rapids_hpo.ipynb to be consistent with the figures and to hopefully surface the rapids aspect first for the sagemaker-examples PR.

File: aws/rapids_sagemaker_hpo.ipynb [hunk 1]
```diff
@@ -0,0 +1,1819 @@
...
+    "         wait = True, logs = 'All')\n",
+    "\n",
+    "hpo.wait() # block until the .fit call above is completed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<span style=\"color:#8735fb; font-size:20pt\"> 3.4 - Results and Summary </span>"
```
Comment (zronaghi) [inline aws/rapids_sagemaker_hpo.ipynb:1411]: This section is a great addition!

File: aws/rapids_sagemaker_hpo.ipynb [hunk 2]
```diff
@@ -0,0 +1,1819 @@
...
+    "\n",
+    "We accelerate HPO in two key ways: \n",
+    "* by *scaling within a node* (e.g., multi-GPU where each GPU brings a magnitude higher core count relative to CPUs), and \n",
+    "* by *scaling across nodes* and running parallel trials on cloud instances.\n",
+    "\n",
+    "By combining these two powers HPO experiments that feel unapproachable and may take multiple days on CPU instances can complete in just hours. For example, we find a <span style=\"color:#8735fb; font-size:14pt\"> **12X** </span> speedup in wall clock time and a <span style=\"color:#8735fb; font-size:14pt\"> **4.5x** </span> reduction in cost when comparing between GPU and CPU [EC2 Spot instances](https://docs.aws.amazon.com/sagemaker/latest/dg/model-managed-spot-training.html) on 100 XGBoost HPO trials using 10 parallel workers on 10 years of the Airline Dataset (~63M flights). For additional details refer to the <a href='#experiments'>end of the notebook</a>."
```
[the remark this answers is not in the document]
Comment (miroenev) [reply aws/rapids_sagemaker_hpo.ipynb:20]: Thanks!

File: aws/rapids_sagemaker_hpo.ipynb [hunk 3]
```diff
@@ -0,0 +1,1819 @@
...
+    "<span style=\"color:#8735fb; font-size:22pt\"> **Key Choices** </span>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's go ahead and choose the configuration options for our HPO run.\n",
+    "\n",
+    "Note that we've set the configuration defaults to values that are on the low end, however you are welcome to scale them up. "
```
[the remark this answers is not in the document]
Comment (miroenev) [reply aws/rapids_sagemaker_hpo.ipynb:61]: I'll bring them back :]

File: aws/rapids_sagemaker_hpo.ipynb (hunk 1 above)
Comment (miroenev) [reply aws/rapids_sagemaker_hpo.ipynb:1411]: Awesome, thanks!

**Pull request merged**